util/mdbook/move_md.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+'''Moves a markdown file, recalculating the relative links for the new location
+
+KNOWN_ISSUES
+- If you move a file to the same dir an existing relative link points to, the path
+  won't be canonicalized (you will get '../<dir>' rather than '.')
+- Doesn't check if link to anchor is correct, just check that file exists
+- Would be nice to re-render the markdown straight from the AST, but I found other
+  formatting changes when trying this.
+'''
+
+import validators
+from markdown_it import MarkdownIt
+from markdown_it.tree import SyntaxTreeNode
+
+import argparse
+import pathlib
+from pathlib import Path
+
+import logging
+logging.basicConfig()
+logging.getLogger().setLevel(logging.DEBUG)  # set root-logger to most-permissive
+logger = logging.getLogger(__name__)
+
+logging.getLogger('markdown_it').setLevel(logging.WARNING)
+
+
+# https://stackoverflow.com/a/71874881
+def relative(target: Path, origin: Path) -> Path:
+    """Return the path of target relative to origin.
+    i.e. --> pathlib equivalent of os.path.relpath'
+    """
+    try:
+        return Path(target).resolve().relative_to(Path(origin).resolve())
+    except ValueError:  # target does not start with origin
+        # recursion with origin (eventually origin is root so try will succeed)
+        return Path('..').joinpath(relative(target, Path(origin).parent))
+
+
+def do_update(l: str, old_path: pathlib.Path) -> bool:
```

Comment:
I would suggest to describe more what the function is doing. Also I am not sure I would call it old_path since this particular function basically just checks that a link points to another file, so it's "unrelated" to being at the old_path vs new_path. Also the function is called do_update but does not actually do anything, maybe call it should_update() or is_link_to_external_file()

Reply:
`should_update()` would be a better name, I will change it. 
Maybe I will pull out the link-check up one level where the name `old_path` has appropriate context, or even remove that code entirely, it was a nice-to-have check when developing the script but really a dedicated link checker (such as `mdbook-linkcheck`) should be run over all the files.